File: src/main/java/io/bloco/faker/components/Address.java
package io.bloco.faker.components;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class Address extends FakerComponent {

    public Address(FakerData data) {
        super(data);
    }

    public String cityPrefix() {
        return fetch("address.city_prefix");
    }

    public String citySuffix() {
        return fetch("address.city_suffix");
    }

    public String country() {
        return fetch("address.country");
    }

    public String countryCode() {
        return fetch("address.country_code");
    }

    public String buildingNumber() {
        return numerify(fetch("address.building_number"));
    }

    public String streetSuffix() {
        return fetch("address.street_suffix");
    }

    public String secondaryAddress() {
        return fetch("address.secondary_address");
    }

    public String postcode() {
        return fetch("address.postcode");
    }

    public String state() {
        return fetch("address.state");
    }

    public String stateAbbr() {
        return fetch("address.state_abbr");
    }

    public String timeZone() {
        return fetch("address.time_zone");
    }

    public String city() {
        return parse(fetch("address.city"));
    }

    public String streetName() {
        return parse(fetch("address.street_name"));
    }

    public String streetAddress() {
        return parse(fetch("address.street_address"));
    }

    public String defaultCountry() {
        return fetch("address.default_country");
    }
}

File: src/main/kotlin/io/bloco/faker/components/Address.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData

class Address(data: FakerData) : FakerComponent(data) {
    fun cityPrefix(): String {
        return fetch("address.city_prefix")
    }

    fun citySuffix(): String {
        return fetch("address.city_suffix")
    }

    fun country(): String {
        return fetch("address.country")
    }

    fun countryCode(): String {
        return fetch("address.country_code")
    }

    fun buildingNumber(): String {
        return numerify(fetch("address.building_number"))
    }

    fun streetSuffix(): String {
        return fetch("address.street_suffix")
    }

    fun secondaryAddress(): String {
        return fetch("address.secondary_address")
    }

    fun postcode(): String {
        return numerify(fetch("address.postcode"))
    }

    fun state(): String {
        return fetch("address.state")
    }

    fun stateAbbr(): String {
        return fetch("address.state_abbr")
    }

    fun timeZone(): String {
        return fetch("address.time_zone")
    }

    fun city(): String {
        return parse(fetch("address.city"))
    }

    fun streetName(): String {
        return parse(fetch("address.street_name"))
    }

    fun streetAddress(): String {
        return parse(fetch("address.street_address"))
    }

    fun defaultCountry(): String {
        return fetch("address.default_country")
    }
}
